cmd/rpc-common.go

```diff
@@ -0,0 +1,107 @@
+/*
+ * Minio Cloud Storage, (C) 2016 Minio, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package cmd
+
+import (
+	"time"
+
+	"github.com/minio/dsync"
+)
+
+// Allow any RPC call request time should be no more/less than 3 seconds.
+// 3 seconds is chosen arbitrarily.
+const rpcSkewTimeAllowed = 3 * time.Second
+
+// AuthRPCArgs represents minimum required arguments to make any authenticated RPC call.
+type AuthRPCArgs struct {
+	// Authentication token to be verified by the server for every RPC call.
+	AuthToken string
+
+	// Request time to be verified by the server for every RPC call.
+	// This is an addition check over Authentication token for time drifting.
+	RequestTime time.Time
+}
+
+// SetAuthToken - sets the token to the supplied value.
+func (args *AuthRPCArgs) SetAuthToken(authToken string) {
+	args.AuthToken = authToken
+}
+
+// SetRequestTime - sets the requestTime to the supplied value.
+func (args *AuthRPCArgs) SetRequestTime(requestTime time.Time) {
+	args.RequestTime = requestTime
+}
+
+// IsAuthenticated - validated whether this auth RPC args are already authenticated or not.
+func (args AuthRPCArgs) IsAuthenticated() error {
+	// Check whether the token is valid
+	if !isAuthTokenValid(args.AuthToken) {
+		return errInvalidToken
+	}
+
+	// Check whether request time is within acceptable skew time.
+	curTime := time.Now().UTC()
+	if curTime.Sub(args.RequestTime) > rpcSkewTimeAllowed || args.RequestTime.Sub(curTime) > rpcSkewTimeAllowed {
```

Comment:
Can you move the second operand to the first one? like how it is done in LoginRPCArgs - IsValid()